components/layout/model.rs

```diff
@@ -446,6 +448,16 @@ pub fn specified_or_none(length: LengthOrPercentageOrNone, containing_length: Au
     }
 }
 
+pub fn specified_or_auto(length: LengthOrPercentageOrAuto, containing_length: Au) -> Au {
+    match length {
+        LengthOrPercentageOrAuto::Length(length) => length,
+        LengthOrPercentageOrAuto::Percentage(p) => containing_length.scale_by(p),
+        LengthOrPercentageOrAuto::Calc(calc) =>
+            containing_length.scale_by(calc.percentage()) + calc.length(),
+        LengthOrPercentageOrAuto::Auto => Au(0),
+    }
+}
+
```

Comment:
This function is unnessasary, and its name is quite confusing, hinting that it will return a `MaybeAuto` type. Use `MaybeAuto::from_style().specified_or_zero()` should be enough.